### PR TITLE
Add nullable annotations to SolutionCrawler files and tiny refactorings

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by tmat** [comment]
@CyrusNajmabadi PTAL

File: src/Features/Core/Portable/SolutionCrawler/IDocumentTrackingService.cs [hunk 1]
```diff
@@ -2,6 +2,8 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/SolutionCrawler/IDocumentTrackingService.cs:5]: Looks like IDocumentTrackingService.TryGetActiveDocument() should return nullable,

File: src/Features/Core/Portable/SolutionCrawler/MiscSolutionCrawlerWorkspaceEventListener.cs [hunk 2]
```diff
@@ -2,6 +2,8 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/SolutionCrawler/MiscSolutionCrawlerWorkspaceEventListener.cs:5]: Is the serviceOpt parameter not marked nullable in IEventListener? Or I think it should then be deriving from `IEventListener<object?>`.

File: src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs [hunk 3]
```diff
@@ -629,18 +648,18 @@ private async Task EnqueueWorkItemForSolutionAsync(Solution solution, Invocation
...
 
                 await EnqueueWorkItemAsync(newProject.GetChanges(oldProject)).ConfigureAwait(continueOnCapturedContext: false);
             }
 
             private async Task EnqueueWorkItemAfterDiffAsync(Solution oldSolution, Solution newSolution, DocumentId documentId)
             {
-                var oldProject = oldSolution.GetProject(documentId.ProjectId);
-                var newProject = newSolution.GetProject(documentId.ProjectId);
+                var oldProject = oldSolution.GetRequiredProject(documentId.ProjectId);
+                var newProject = newSolution.GetRequiredProject(documentId.ProjectId);
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs:660]: No worry that this gets called during added/removed operations and the nulls were allowed before?

File: src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerLogger.cs [hunk 4]
```diff
@@ -236,9 +238,9 @@ public static void LogIncrementalAnalyzerProcessorStatistics(int correlationId,
 
                     m[CreateProperty(propertyName, Max)] = result.Maximum;
                     m[CreateProperty(propertyName, Min)] = result.Minimum;
-                    m[CreateProperty(propertyName, Median)] = result.Median.Value;
+                    m[CreateProperty(propertyName, Median)] = result.Median!.Value;
```
Comment (jasonmalinowski) [inline src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerLogger.cs:241]: What's going on here?

File: src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs (hunk 3 above)
Comment (tmat) [reply src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs:660]: It would have thrown NRE if the projects didn't exist.

Comment (tmat) [reply src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs:660]: The docs are also required by EnqueueWorkItemAsync .

File: src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerLogger.cs (hunk 4 above)
Comment (tmat) [reply src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerLogger.cs:241]: StatisticResult may or may not have Median calculated. In this case it does.

**Pull request merged**